### ISSUE TITLE
Package operations: 1 update - but there is none

### ISSUE THREAD

**Issue by Commifreak** [composer/composer#6796]
My `composer.json`:

```json
{
  "name": "yiisoft/yii2-app-advanced",
  "description": "Yii 2 Advanced Project Template",
  "keywords": [
    "yii2",
    "framework",
    "advanced",
    "project template"
  ],
  "homepage": "http://www.yiiframework.com/",
  "type": "project",
  "license": "BSD-3-Clause",
  "support": {
    "issues": "https://github.com/yiisoft/yii2/issues?state=open",
    "forum": "http://www.yiiframework.com/forum/",
    "wiki": "http://www.yiiframework.com/wiki/",
    "irc": "irc://irc.freenode.net/yii",
    "source": "https://github.com/yiisoft/yii2"
  },
  "minimum-stability": "dev",
  "prefer-stable": true,
  "require": {
    "php": ">=5.4.0",
    "yiisoft/yii2": "~2.0.6",
    "yiisoft/yii2-bootstrap": "~2.0.0",
    "yiisoft/yii2-swiftmailer": "~2.0.0",
    "samdark/yii2-webshell": "~1.0",
    "kartik-v/yii2-widget-sidenav": "*",
    "kartik-v/yii2-widget-activeform": "*",
    "kartik-v/yii2-widgets": "*",
    "kartik-v/yii2-tabs-x": "*",
    "kartik-v/yii2-detail-view": "*",
    "kartik-v/yii2-widget-growl": "*",
    "kartik-v/yii2-nav-x": "*",
    "yiisoft/yii2-imagine": "^2.1",
    "kartik-v/yii2-grid": "@dev",
    "kartik-v/yii2-mpdf": "*",
    "kartik-v/yii2-editable": "*",
    "phpoffice/phpexcel": "^1.8",
    "2amigos/yii2-file-upload-widget": "*",
    "moonlandsoft/yii2-phpexcel": "*",
    "kartik-v/yii2-dynagrid": "@dev",
    "liyunfang/yii2-widget-linkpager": "^1.0",
    "franciscomaya/yii2-sceditor": "dev-master",
    "newerton/yii2-fancybox": "dev-master",
    "kartik-v/yii2-icons": "^1.4"
  },
  "require-dev": {
    "yiisoft/yii2-debug": "~2.0.0",
    "yiisoft/yii2-gii": "~2.0.0",
    "yiisoft/yii2-faker": "~2.0.0",
    "codeception/base": "^2.2.3",
    "codeception/verify": "~0.3.1"
  },
  "config": {
    "process-timeout": 1800
  },
  "extra": {
    "asset-installer-paths": {
      "npm-asset-library": "vendor/npm",
      "bower-asset-library": "vendor/bower"
    }
  },
  "scripts": {
    "post-install-cmd": "php init --env=Development --overwrite=n"
  }
}

```

Output of `composer diagnose`:

```
Checking composer.json: WARNING
require.kartik-v/yii2-widget-sidenav : unbound version constraints (*) should be avoided
require.kartik-v/yii2-widget-activeform : unbound version constraints (*) should be avoided
require.kartik-v/yii2-widgets : unbound version constraints (*) should be avoided
require.kartik-v/yii2-tabs-x : unbound version constraints (*) should be avoided
require.kartik-v/yii2-detail-view : unbound version constraints (*) should be avoided
require.kartik-v/yii2-widget-growl : unbound version constraints (*) should be avoided
require.kartik-v/yii2-nav-x : unbound version constraints (*) should be avoided
require.kartik-v/yii2-grid : unbound version constraints (@dev) should be avoided
require.kartik-v/yii2-mpdf : unbound version constraints (*) should be avoided
require.kartik-v/yii2-editable : unbound version constraints (*) should be avoided
require.2amigos/yii2-file-upload-widget : unbound version constraints (*) should be avoided
require.moonlandsoft/yii2-phpexcel : unbound version constraints (*) should be avoided
require.kartik-v/yii2-dynagrid : unbound version constraints (@dev) should be avoided
require.franciscomaya/yii2-sceditor : unbound version constraints (dev-master) should be avoided
require.newerton/yii2-fancybox : unbound version constraints (dev-master) should be avoided
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking HTTP proxy: OK
Checking HTTP proxy support for request_fulluri: OK
Checking HTTPS proxy support for request_fulluri: OK
Checking github.com oauth access: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
```

When I run this command:

```
composer update
```

I get the following output:

```
The "extra.asset-installer-paths" option is deprecated, use the "config.fxp-asset.installer-paths" option
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 1 update, 0 removals
Generating autoload files
```

And I expected this to happen: composer list 1 update, but does nothing. --dry-run shows the same. Whats that mysterious 1 update?

**Comment by alcohol** [comment]
Can you see if the same happens when you run with `--no-plugins`?

**Comment by Commifreak** [comment]
Then this appears:

```
composer update --no-plugins
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - newerton/yii2-fancybox dev-master requires bower-asset/jquery-mousewheel >=3.1.13 -> no matching package found.
    - newerton/yii2-fancybox dev-master requires bower-asset/jquery-mousewheel >=3.1.13 -> no matching package found.
    - Installation request for newerton/yii2-fancybox dev-master -> satisfiable by newerton/yii2-fancybox[dev-master].

Potential causes:
 - A typo in the package name
 - The package is not available in a stable-enough version according to your minimum-stability setting
   see <https://getcomposer.org/doc/04-schema.md#minimum-stability> for more details.
 - It's a private package and you forgot to add a custom repository to find it

Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems.
```

I try to get further ;)

**Comment by alcohol** [comment]
Instructions unclear, cannot reproduce;

```
rob@galaga /tmp $ mkdir foo bar
mkdir: created directory 'foo'
mkdir: created directory 'bar'
rob@galaga /tmp $ cd foo
rob@galaga /tmp/foo $ composer require symfony/console ~4.0.0
./composer.json has been created
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 2 installs, 0 updates, 0 removals
  - Installing symfony/polyfill-mbstring (v1.9.0): Loading from cache
  - Installing symfony/console (v4.0.14): Downloading (100%)         
symfony/console suggests installing psr/log-implementation (For using the console logger)
symfony/console suggests installing symfony/event-dispatcher
symfony/console suggests installing symfony/lock
symfony/console suggests installing symfony/process
Writing lock file
Generating autoload files
rob@galaga /tmp/foo $ cp composer.* ../bar
rob@galaga /tmp/foo $ cd ../bar
rob@galaga /tmp/bar $ composer install
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Package operations: 2 installs, 0 updates, 0 removals
  - Installing symfony/polyfill-mbstring (v1.9.0): Loading from cache
  - Installing symfony/console (v4.0.14): Loading from cache
symfony/console suggests installing psr/log-implementation (For using the console logger)
symfony/console suggests installing symfony/event-dispatcher
symfony/console suggests installing symfony/lock
symfony/console suggests installing symfony/process
Generating autoload files
rob@galaga /tmp/bar $ cd ../foo
rob@galaga /tmp/foo $ composer require symfony/console ~4.1.0
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 1 update, 0 removals
  - Updating symfony/console (v4.0.14 => v4.1.4): Loading from cache
Writing lock file
Generating autoload files
rob@galaga /tmp/foo $ cp composer.* ../bar
cp: overwrite '../bar/composer.json'? y
cp: overwrite '../bar/composer.lock'? y
rob@galaga /tmp/foo $ cd ../bar
rob@galaga /tmp/bar $ composer install
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Package operations: 0 installs, 1 update, 0 removals
  - Updating symfony/console (v4.0.14 => v4.1.4): Loading from cache
Generating autoload files
```

**Comment by alcohol** [comment]
The issue regarding the install/update/removal action counts has been raised before, so it seems very likely there is something off in the logic behind it. However, since the impact is merely visual (when it was added, the actual solving processes and such were not modified), not much time has yet been invested into fully investigating what is going on.

**Comment by Seldaek** [comment]
Closing as hopefully fixed by 2.0, if not please reopen.
